qubesmanager/global_settings.py

```diff
@@ -36,6 +37,52 @@
 
 qmemman_config_path = '/etc/qubes/qmemman.conf'
 
+def _run_qrexec_repo(service, arg=''):
+    # Fake up a "qrexec call" to dom0 because dom0 can't qrexec to itself yet
+    cmd = '/etc/qubes-rpc/' + service
+    p = subprocess.run(
+        ['sudo', cmd, arg],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE
+    )
+    assert not p.stderr
+    assert p.returncode == 0
+    return p.stdout.decode('utf-8')
+
+def _manage_repos(repolist, action):
+    for i in repolist:
+        result = _run_qrexec_repo('qubes.repos.' + action, i)
+        assert result == 'ok\n'
+
+def _handle_dom0_updates_combobox(idx):
+    idx += 1
+    repolist = ['qubes-dom0-current', 'qubes-dom0-security-testing',
+         'qubes-dom0-current-testing', 'qubes-dom0-unstable']
+    enable = repolist[:idx]
+    disable = repolist[idx:]
+    _manage_repos(enable, 'Enable')
+    _manage_repos(disable, 'Disable')
```

Comment:
Error in all those functions should result in readable error message. Something like "failed to enable/disable repository x" (which may be a text of raised exception inside _manage_repos). But it should be shown to the user in readable form, not generic "Huston, we have a problem...". Generally that handler is last resort, meaning some error handling is broken. See `VMSettingsWindow.save_changes` in `settings.py`. Other settings in this file lack proper error handling, so don't use them as an example...